### PR TITLE
Use public_key on NEAR

### PR DESCRIPTION
**Pull request by pkoch**
https://app.asana.com/0/1204479969653186/1205885384119890/f

### CODE REVIEW

File: near-rs/integration-tests/src/tests.rs [hunk 1]
```diff
@@ -3,18 +3,39 @@ use std::{
...
     locked_until: u128,
 }
 
+async fn create_account(new_account_id: &str, worker: &Worker<Sandbox>) -> anyhow::Result<Account> {
+    Ok(worker.dev_create_account().await?
+        .create_subaccount(new_account_id)
+        .transact()
+        .await?
+        .into_result()?)
+}
```
Comment (pkoch) [inline near-rs/integration-tests/src/tests.rs:24]: This doesn't seem to be working as I expect. I get all the accounts with the same signer.

Comment (pkoch) [reply near-rs/integration-tests/src/tests.rs:24]: Nevermind, I realized I don't need full account, only keys. Generated random ones.

File: near-rs/contract/src/lib.rs [hunk 2]
```diff
@@ -2,31 +2,36 @@ extern crate near_sdk;
...
     data_id: String,
     locked_until: EpochHeight,
 }
 
 pub fn derive_grant_id(grant: &Grant) -> String {
     let id = format!(
         "{}{}{}{}",
-        grant.owner, grant.grantee, grant.data_id, grant.locked_until,
+        // FIXME: What's the right way to serialize this without the '"'s?
+        to_string(&grant.owner).unwrap().replace("\"", ""),
```
Comment (pkoch) [inline near-rs/contract/src/lib.rs:31]: @juliosantos have you perhaps learned how to work with serde? I'd like to get rid of this ugly hack.

`PublicKey` is not `Display`. It is `Debug`, but it feels wrong to rely on that.

Comment (pkoch) [reply near-rs/contract/src/lib.rs]: Júlio fixed this with his own commits.

**Comment by pkoch** [comment]
@juliosantos sanity check ship?

File: near-rs/contract/src/lib.rs [hunk 3]
```diff
@@ -2,31 +2,46 @@ extern crate near_sdk;
...
-    grantee: AccountId,
+    owner: PublicKey,
+    grantee: PublicKey,
     data_id: String,
     locked_until: EpochHeight,
 }
 
+#[cfg(test)]
+#[test]
+fn derive_grant_id_example() {
```
Comment (juliosantos) [inline near-rs/contract/src/lib.rs:28]: omg ty

File: near-rs/contract/src/lib.rs [hunk 4]
```diff
@@ -2,31 +2,46 @@ extern crate near_sdk;
...
+        })
+    )
+}
+
 pub fn derive_grant_id(grant: &Grant) -> String {
     let id = format!(
-        "{}{}{}{}",
-        grant.owner, grant.grantee, grant.data_id, grant.locked_until,
+        "{:?}{:?}{}{}",
+        &grant.owner, &grant.grantee, grant.data_id, grant.locked_until,
```
Comment (juliosantos) [inline near-rs/contract/src/lib.rs:44]: Does this need to be a reference?

Comment (pkoch) [reply near-rs/contract/src/lib.rs:44]: Good catch.

**Review by juliosantos — APPROVED** [review]

**Pull request merged**